src/Illuminate/Validation/Rules/RegExp.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Illuminate\Validation\Rules;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Illuminate\Support\Str;
+use Illuminate\Support\Stringable as SupportStringable;
+use Stringable;
+
+class RegExp implements Stringable
+{
+    protected string $regExp;
+    protected bool $negated = false;
+
+    /**
+     * @var string[]
+     */
+    protected array $flags = [];
+
+    /**
+     * Create a new array rule instance.
+     *
+     * @param  string  $regExp
+     * @param  string[]|\Illuminate\Contracts\Support\Arrayable|null  $flags
+     * @return void
+     */
+    public function __construct(string $regExp, $flags = null)
+    {
+        $this->regExp = $regExp;
+        $this->flags(...$flags);
+    }
+
+    public function active(bool $active)
+    {
+        $this->active = $active;
+
+        return $this;
+    }
+
+    /**
+     * @param  string[]|\Illuminate\Contracts\Support\Arrayable|null  $flags
+     */
+    public function flags($flags = null)
```

Comment:
I feel like accepting `null` here and in the constructor is counterintuitive, especially if it is an invalid input. What if someone calls `->flags(null)`, wouldn't your code assign `$this->flags` to be `[null]`?

I propose `array|Arrayable $flags = []` for this method and the constructor. Or, to allow `func_get_args()` to work, `$flags = []`.

Reply:
The class essentially has two methods:
* `setFlags()`: Overwrites/replaces the existing flags with a set of new ones (here, it's called `flags()` due to naming conventions)
* `addFlag()`: Adds a flag to the set of existing flags without replacing them (here, it's called `flag()` due to naming conventions)